packages/cosmo-dycore/package.py

```diff
@@ -139,5 +139,8 @@ def cmake_args(self):
     @run_after('install')
     @on_package_attributes(run_tests=True)
     def test(self):
-      if '+build_tests' in self.spec: 
-            subprocess.run(['./test_dycore.py',  str(self.spec),  self.build_directory], cwd = self.root_cmakelists_dir + '/test/tools')
+      if '+build_tests' in self.spec:
+          try:
+              subprocess.run(['./test_dycore.py',  str(self.spec),  self.build_directory], cwd = self.root_cmakelists_dir + '/test/tools', check=True, stderr=subprocess.STDOUT)
+          except subprocess.CalledProcessError:
```

Comment:
```suggestion
          except:
```
This will catch other exceptions too that could lead to false positives.